### ISSUE TITLE
Can't handle large files (>2GB) on Windows

### ISSUE THREAD

**Issue by zzcoder** [github/git-lfs#1218]
The windows git-lfs gives this error when adding large files,

```
C:\Source\test-data>git add src\test\resources\test.bin
fatal: Out of memory? mmap failed: No such file or directory
```

The test.bin is 3GB.  This is in on Windows 10 64-bit. The same command works on other platforms (Centos 7, OS X).

```
git lfs version
git-lfs/1.2.0 (GitHub; windows amd64; go 1.6.1; git 386c5d8)

```

**Comment by technoweenie** [comment]
This may be dependent on your Git version. What versions are you running on Windows, Centos 7, and OSX?

**Comment by paulstelian97** [comment]
I wonder why git even uses mmap. Looks like an address space issue (the git executable is 32 bits). The error message reported by perror/strerror looks weird/misdirecting.

**Comment by technoweenie** [comment]
Looks like this was [fixed in Git 2.2.0](https://github.com/git/git/blob/v2.2.0/Documentation/RelNotes/2.2.0.txt#L162-L164). I just added 2 3GB files on my home Windows machine with Git LFS v1.2.0 and Git v2.7.1. The memory usage stayed constant around 2MB.

@zzcoder: if you're still having problems, please post more details about your setup. Thanks!
